applications/StructuralMechanicsApplication/python_scripts/adaptative_remeshing_structural_mechanics_analysis.py

```diff
@@ -25,48 +25,60 @@ def __init__(self, model, project_parameters):
             "analysis_type" : "linear"
         }
         """)
-        if (project_parameters["solver_settings"].Has("max_iteration") is True):
+        if project_parameters["solver_settings"].Has("max_iteration"):
             self.non_linear_iterations = project_parameters["solver_settings"]["max_iteration"].GetInt()
         else:
             self.non_linear_iterations = 10
             project_parameters["solver_settings"].AddValue("max_iteration", default_params["max_iteration"])
-        if (project_parameters["solver_settings"].Has("analysis_type") is True):
+        if project_parameters["solver_settings"].Has("analysis_type"):
             project_parameters["solver_settings"]["analysis_type"].SetString("linear")
         else:
             project_parameters["solver_settings"].AddValue("analysis_type", default_params["analysis_type"])
-        if (project_parameters.Has("recursive_remeshing_process") is True):
+        self.process_remesh = False
+        if project_parameters.Has("recursive_remeshing_process"):
             self.process_remesh = True
-        else:
-            self.process_remesh = False
+        if project_parameters.Has("processes"):
+            if project_parameters["processes"].Has("recursive_remeshing_process"):
+                self.process_remesh = True
         super(AdaptativeRemeshingStructuralMechanicsAnalysis, self).__init__(model, project_parameters)
 
     def Initialize(self):
         """ Initializing the Analysis """
         super(AdaptativeRemeshingStructuralMechanicsAnalysis, self).Initialize()
-        if (self.process_remesh is False):
+        computing_model_part = self._GetSolver().GetComputingModelPart()
+        if not self.process_remesh:
             convergence_criteria = self._GetSolver().get_convergence_criterion()
-            convergence_criteria.Initialize(self._GetSolver().GetComputingModelPart())
+            convergence_criteria.Initialize(computing_model_part)
+
         # Ensuring to have conditions on the BC before remesh
-        computing_model_part = self._GetSolver().GetComputingModelPart()
-        # We need to detect the conditions in the boundary conditions
-        if (self.project_parameters.Has("constraints_process_list") is True):
-            constraints_process_list = self.project_parameters["constraints_process_list"]
+        is_surface = False
+        for elem in computing_model_part.Elements:
+            geom = elem.GetGeometry()
+            if geom.WorkingSpaceDimension() != geom.LocalSpaceDimension():
+                is_surface = True
+            break
```

Comment:
thisis implemened like four times or so, please refactor to athe common-methods class

Reply:
Well, I can create a very very small utility

Reply:
hm, you know better. Maybe its not worth it

Reply:
I think not sincerely